source/apple/ipc-socket-osx.cpp

```diff
@@ -97,13 +101,25 @@ uint32_t os::apple::socket_osx::write(const char *buffer, size_t buffer_length,
 {
     os::error err = os::error::Error;
     int ret = 0;
+    int sizeChunks = 8*1024; // 8KB
     int file_descriptor = t == REQUEST ? file_req : file_rep;
-    std::string typePipe = t == REQUEST ? "request" : "reply";
+    std::string typePipe = t == REQUEST ? "client" : "server";
 
     if (file_descriptor < 0)
         goto end;
 
-    ret = ::write(file_descriptor, buffer, buffer_length);
+    if (buffer_length <= sizeChunks) {
+        ret = ::write(file_descriptor, buffer, buffer_length);
+    } else {
+        int size_wrote = 0;
+        while (size_wrote < buffer_length) {
+            bool end = (buffer_length - size_wrote) <= sizeChunks;
+            int size_to_write = end ? buffer_length - size_wrote : sizeChunks;
+            ret = ::write(file_descriptor, buffer + size_wrote, size_to_write);
```

Comment:
if we not check ret here then next chunk write can set it to positive again and it hide failed write.